src/SignalR/clients/csharp/Client/test/UnitTests/ServerSentEventsParserTests.cs

```diff
@@ -102,6 +95,10 @@ public void ParseSSEMessageIncompleteParseResult(string encodedMessage)
     [InlineData(new[] { "data: Hello, World", "\r\n\r\n" }, "Hello, World")]
     [InlineData(new[] { "data: Hello, World\r\n", "\r\n" }, "Hello, World")]
     [InlineData(new[] { "data: ", "Hello, World\r\n\r\n" }, "Hello, World")]
+    [InlineData(new[] { "data: ", "Hello, World\r\r" }, "Hello, World")]
+    [InlineData(new[] { "data: ", "Hello, World\n\n" }, "Hello, World")]
+    [InlineData(new[] { "data: ", "Hello, World\r\n\r" }, "Hello, World")]
+    [InlineData(new[] { "data: ", "Hello, World\r\n\n" }, "Hello, World")]
```

Comment:
We have no way to distinguish between messages when reading. That means
`"data: ", "Hello\r", "\n"` would behave differently than `"data: ", "Hello\r\n"`

This is on the client, so it's less concerning. But it does feel like we're making the client flaky. I wonder if we can compromise and accept `\r\n` or `\n` and not support just `\r`.

Reply:
Adding to that: if the buffer would only contain up to \\r of an \\r\\n the data would be mangled up.

Either we have to absolutely read the next byte (s) to ensure a complete line (more complexity) or just accept that hopefully people either use Unix or windows line endings.

This would mean that we also need to adjust the docs to clarify that we only support a subset of the SSE line endings.

Reply:
> Either we have to absolutely read the next byte (s) to ensure a complete line (more complexity)

I think this is what the spec is referring to here:

"while line buffering with lines are defined to end with a single U+000A LINE FEED (LF) character is safe, block buffering or line buffering with different expected line endings can cause delays in event dispatch."

I think adding `\n` to our current support is good enough. And we should update the SignalR spec.

Reply:
Do you think the implementation should trim any trailing `\r`?
for cases like `data: \r\r\n`, which would result in `data: \r`